### DISCUSSION TITLE
[Prettier Plugin] No sorting for `px-4 py-2`?

### DISCUSSION THREAD

**Discussion by brc-dd** [tailwindlabs/tailwindcss#8689] [Ideas]
Try tidying this: https://play.tailwindcss.com/HQlI8YhkrG

Result:

```html
<div class="px-4 py-2">Hey There</div>
<div class="py-2 px-4">Hey There</div>
```

These classes don't get sorted?

**Comment by tofuwu-dev** [comment]
Hi @brc-dd 
If you use prettier formatter, you can install `prettier` and `prettier prettier-plugin-tailwindcss` for sorting tailwind class
```bash
npm install -D prettier prettier-plugin-tailwindcss
```
https://tailwindcss.com/blog/automatic-class-sorting-with-prettier

**Reply by brc-dd** [reply]
Hi. I'm using that plugin only. TailwindCSS Playground also uses it along with prettier. But that plugin is not sorting this specific case.

**Reply by tofuwu-dev** [reply]
The plugin works good with Prettier Code Formatter on vs-code by just pressing `ctrl + s`.
```bash
# Before
<button class="text-white px-4 sm:px-8 py-2 sm:py-3 bg-sky-700 hover:bg-sky-800">...</button>

# After
<button class="bg-sky-700 px-4 py-2 text-white hover:bg-sky-800 sm:px-8 sm:py-3">...</button>
```
But not sure on Playground.. It would not be worked

**Reply by brc-dd** [reply]
Try sorting this:

```html
<button class="text-white py-2 sm:py-3 px-4 sm:px-8 bg-sky-700 hover:bg-sky-800">...</button>
```

It'll give you this:

```html
<button class="bg-sky-700 py-2 px-4 text-white hover:bg-sky-800 sm:py-3 sm:px-8">...</button>
```

Difference between what your text gives after formatting and this one:

![image](https://user-images.githubusercontent.com/40380293/174655828-4117b3e5-5dec-44bd-a0d2-ca317da854af.png)

Those classnames aren't getting sorted. Running prettier does bring them side by side, but preserves the original order they were written in.

**Reply by tofuwu-dev** [reply]
Oh, weird. I got such an case... It seems like that prettier doesnt work properly.
We will need better plugin for sorting class...
